### PR TITLE
Merge v2 into master

### PR DESCRIPTION
**Pull request by bkrausz**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,106 +1,116 @@
-=== 1.18.0 2015-01-21
+<<<<<<< HEAD:CHANGELOG.md
```
Comment (GrahamCampbell) [inline CHANGELOG.md]: Ooops...

File: README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# Stripe PHP bindings
+
+[![Build Status](https://travis-ci.org/stripe/stripe-php.svg?branch=master)](https://travis-ci.org/stripe/stripe-php) 
+[![Latest Stable Version](https://poser.pugx.org/stripe/stripe-php/v/stable.svg)](https://packagist.org/packages/stripe/stripe-php)
+[![Total Downloads](https://poser.pugx.org/stripe/stripe-php/downloads.svg)](https://packagist.org/packages/stripe/stripe-php)
+[![License](https://poser.pugx.org/stripe/stripe-php/license.svg)](https://packagist.org/packages/stripe/stripe-php)
+[![Code Coverage](https://coveralls.io/repos/stripe/stripe-php/badge.png?branch=experimental-v2)](https://coveralls.io/r/stripe/stripe-php?branch=experimental-v2)
+
+
+
```
Comment (GrahamCampbell) [inline README.md]: Why so many new lines?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
+# Stripe PHP bindings
+
+[![Build Status](https://travis-ci.org/stripe/stripe-php.svg?branch=master)](https://travis-ci.org/stripe/stripe-php) 
+[![Latest Stable Version](https://poser.pugx.org/stripe/stripe-php/v/stable.svg)](https://packagist.org/packages/stripe/stripe-php)
+[![Total Downloads](https://poser.pugx.org/stripe/stripe-php/downloads.svg)](https://packagist.org/packages/stripe/stripe-php)
+[![License](https://poser.pugx.org/stripe/stripe-php/license.svg)](https://packagist.org/packages/stripe/stripe-php)
+[![Code Coverage](https://coveralls.io/repos/stripe/stripe-php/badge.png?branch=experimental-v2)](https://coveralls.io/r/stripe/stripe-php?branch=experimental-v2)
```
Comment (GrahamCampbell) [inline README.md]: Bages need uodating.

File: VERSION [hunk 4]
```diff
@@ -1 +1 @@
-1.18.0
+2.0.0
```
Comment (GrahamCampbell) [inline VERSION]: Why do we have this file?

File: composer.json [hunk 5]
```diff
@@ -15,15 +15,17 @@
...
   },
   "require-dev": {
-    "simpletest/simpletest": "*"
+    "phpunit/phpunit": "~4.0",
+    "satooshi/php-coveralls": "~0.6.1",
+    "squizlabs/php_codesniffer": "~2.0"
   },
   "autoload": {
-    "classmap": ["lib/Stripe/"]
+    "psr-4": { "Stripe\\" : "lib" }
```
Comment (GrahamCampbell) [inline composer.json]: missing a `/` on the end of `lib`.

File: lib/Account.php [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+namespace Stripe;
+
+class Account extends ApiResource
+{
+    public function instanceUrl()
+    {
+        if ($this['id'] === null) {
+            return '/v1/account';
+        } else {
```
Comment (GrahamCampbell) [inline lib/Account.php]: Else block not needed.

Comment (bkrausz) [inline lib/Account.php]: Subjective style decision, either way is practically fine (unless PSR-2 has an opinion here).

Comment (GrahamCampbell) [inline lib/Account.php]: It's not a cs thing. It's just a micro-optimization.

**Comment by GrahamCampbell** [comment]
For your interest, just run this through my cs checker: https://dev.styleci.io/commits/d2940754c09a7f08ff48a0595b63420b5b768ff6.

**Comment by GrahamCampbell** [comment]
Ignore the short array syntax changes.

**Comment by GrahamCampbell** [comment]
Also, I think it would be worth only supporting php 5.4+ since both php 5.2/5.3 are eol, there's no point in supporting 5.3 still.

File: VERSION (hunk 4 above)
Comment (bkrausz) [inline VERSION]: I suspect part of Stripe's automatic hosting of tags on our servers. @ebroder is this no longer in use?

**Comment by bkrausz** [comment]
Almost all of those are bikesheds around things like spacing around `.` and tab alignment. Unless you're referencing a spec we've agreed to follow, I don't think it's worth changing about 10% of the codebase to align with that particular checker's rules.

Re: 5.3, this was discussed at #111 (and in a number of other issues). Given that packagist is still seeing about 15% usage from 5.3 (which I believe lines up with the 5.3 merchant count on Stripe itself), we're going to support 5.3 for the time being. If, in a few months, the number of 5.3s on the v2.* version of the bindings is quite small, we will know that those folks are unlikely to upgrade anyway, and cut a v3.0 that drops 5.3 support.

**Comment by GrahamCampbell** [comment]
> align with that particular checker's rules.

StyleCI is actually configurable with a .php_cs file. It's just set for psr-2 plus a few extra rules that make sense imo. Many are from symfony's coding style guide.

**Comment by GrahamCampbell** [comment]
I just put that there for interest. That's why I haven't send a pull to change anything.

File: VERSION (hunk 4 above)
Comment (ebroder) [inline VERSION]: Yeah, our tooling for code.stripe.com wants to have a consistent cross-language way of detecting versions, and this is what we use for that.

**Pull request merged**